examples/chainable_router.rs

```diff
@@ -0,0 +1,22 @@
+#[macro_use] extern crate nickel;
+extern crate hyper;
+
+use nickel::{Nickel};
+use nickel::router::chainable_router::{ChainablePath, ChainableHandler};
+
+fn main() {
+    let mut server = Nickel::new();
+    server
+        .route("/")
+        .get(middleware!("get"))
+        .post(middleware!("post"))
+        .put(middleware!("put"))
+        .patch(middleware!("patch"))
+        .delete(middleware!("delete"))
+        .route("/chain1")
```

Comment:
I'm unsure about chaining `route` here, it feels a bit at odds with rust's ownership model here, where there's a non-obvious change to the receiver in the middle of the chain.

I find it a lot less clear than splitting it out into:
```rust
server.route("/")
      .get(...)
      .post(..);

server.route("/chain1")
      .get(..)
      .post(..);
```

Perhaps it's worth not having the feature at first and then seeing if the ask appears in future?